src/search/landmarks/landmark_factory.cc

```diff
@@ -339,14 +339,18 @@ void LandmarkFactory::approximate_reasonable_orders(
         if (node_p->disjunctive)
             continue;
 
-        if (node_p->is_true_in_state(initial_state))
-            return;
+        if (obedient_orders && node_p->is_true_in_state(initial_state))
+            continue;
 
         if (!obedient_orders && node_p->is_true_in_goal) {
             for (auto &node2_p : lm_graph->get_nodes()) {
                 if (node2_p == node_p || node2_p->disjunctive)
                     continue;
-                if (interferes(task_proxy, node2_p.get(), node_p.get())) {
+                if (node_p->is_true_in_state(initial_state)
```

Comment:
I find it nicer to either catch all three "continue" cases by themselves or all at once (and I prefer to catch them all at once). The 2-to-1 split seems arbitrary.

Reply:
A similar comment to what I wrote about the common achievers applies here: should we only fix the bug or make the implementation more reasonable (no pun intended)?
Option 1: Keep the check `node_p->is_true_in_state(initial_state)` where it was, right at the start of the loop, and replace the `return` with a `continue`.
Option 2: Do it like it's implemented in this patch to allow orders A->B where B is true initially.

On the one side, option 1 does preserve the semantics of the current code, but does not terminate too early anymore. On the other hand, I see no good argument for disregarding orderings A->B where B is initially true, so we might as well change to option 2 in this issue.